internal/cli/cmd/create/credentials.go

```diff
@@ -0,0 +1,204 @@
+	option.Image(cmd.Flags(), &o.Image, "Create credentials for a container image repository.")
+	option.RepoURL(cmd.Flags(), &o.RepoURL, "URL of the repository the credentials are for.")
+	option.RepoURLPattern(
+		cmd.Flags(), &o.RepoURLPattern,
+		"Regular expression matching multiple repositories the credentials are for.",
+	)
+	option.Username(cmd.Flags(), &o.Username, "Username for the credentials.")
+	option.Password(cmd.Flags(), &o.Password, "Password for the credentials.")
+
+	cmd.MarkFlagsOneRequired(option.GitFlag, option.HelmFlag, option.ImageFlag)
+	cmd.MarkFlagsMutuallyExclusive(option.GitFlag, option.HelmFlag, option.ImageFlag)
+
+	cmd.MarkFlagsOneRequired(option.RepoURLFlag, option.RepoURLPatternFlag)
+	cmd.MarkFlagsMutuallyExclusive(option.RepoURLFlag, option.RepoURLPatternFlag)
+
+	if err := cmd.MarkFlagRequired(option.UsernameFlag); err != nil {
+		panic(errors.Wrapf(err, "could not mark %s flag as required", option.UsernameFlag))
+	}
+}
+
+// complete sets the options from the command arguments.
+func (o *createCredentialsOptions) complete(args []string) {
+	o.Name = args[0]
+}
+
+// validate performs validation of the options. If the options are invalid, an
+// error is returned.
+func (o *createCredentialsOptions) validate() error {
+	var errs []error
+	// While the flags are marked as required, a user could still provide an empty
+	// string. This is a check to ensure that the flags are not empty.
+	if o.Project == "" {
+		errs = append(errs, errors.New("project is required"))
+	}
+	if o.RepoURL == "" && o.RepoURLPattern == "" {
+		errs = append(
+			errs,
+			errors.New("either repo-url or repo-url-pattern is required"),
+		)
+	}
+	if o.Username == "" {
+		errs = append(errs, errors.New("username is required"))
+	}
+	return goerrors.Join(errs...)
+}
+
+// run creates the credentials in the project based on the options.
+func (o *createCredentialsOptions) run(ctx context.Context) error {
+	for {
+		if o.Password != "" {
+			break
+		}
+		prompt := &survey.Password{
+			Message: "Repository password",
+		}
+		if err := survey.AskOne(prompt, &o.Password); err != nil {
+			return err
+		}
+	}
+
+	kargoSvcCli, err := client.GetClientFromConfig(ctx, o.Config, o.Option)
+	if err != nil {
+		return errors.Wrap(err, "get client from config")
+	}
+
+	var repoType string
+	if o.Git {
+		repoType = string(credentials.TypeGit)
+	} else if o.Helm {
+		repoType = string(credentials.TypeHelm)
+	} else if o.Image {
+		repoType = string(credentials.TypeImage)
+	}
+
+	if _, err := kargoSvcCli.CreateCredentials(
+		ctx,
+		connect.NewRequest(
+			&v1alpha1.CreateCredentialsRequest{
+				Project:        o.Project,
+				Name:           o.Name,
+				Type:           repoType,
+				RepoUrl:        o.RepoURL,
+				RepoUrlPattern: o.RepoURLPattern,
+				Username:       o.Username,
+				Password:       o.Password,
+			},
+		),
+	); err != nil {
+		return errors.Wrap(err, "create credentials")
+	}
+
+	return nil
```

Comment:
We should probably print here what we created, see the project command for further inspiration on how to do this with recent changes made to `main`:

https://github.com/akuity/kargo/blob/34bcb82cb0696c3b390e3321f786ff21c2702071/internal/cli/cmd/create/project.go#L128-L132

Reply:
Yeah... I realized this while in the midst of merging the latest from main into here. I plan to.

Reply:
I've added this in e7fc174a7c92447e1095d8d8878c603ac0bfe3b5 but it's still done the old way...

I'll merge shortly and will make sure everything is consistent with your recent changes.

Reply:
Fixed in c73f394aca28048395ae334e36b655467ba105a2